src/Sulu/Component/Content/Document/Subscriber/RouteSubscriber.php

```diff
@@ -38,10 +72,62 @@ public function handleMetadataLoad(MetadataLoadEvent $event)
             return;
         }
 
-        $metadata->addFieldMapping('targetDocument', [
-            'encoding' => 'system',
-            'property' => self::DOCUMENT_TARGET_FIELD,
-            'type' => 'reference',
-        ]);
+        $metadata->addFieldMapping(
+            'created',
+            [
+                'encoding' => 'system',
+                'property' => self::CREATED,
+                'type' => 'date',
+            ]
+        );
+    }
+
+    public function handlePersist(PersistEvent $event)
+    {
+        $document = $event->getDocument();
+
+        if (!$document instanceof RouteBehavior) {
+            return;
+        }
+
+        $node = $event->getNode();
+        $session = $node->getSession();
+
+        /** @var PageDocument $targetDocument */
+        $targetDocument = $document->getTargetDocument();
+        $targetNode = $this->documentInspector->getNode($targetDocument);
+
+        // copy new route to old position
+        $webspaceKey = $targetDocument->getWebspaceName();
```

Comment:
When looking at the implementation of the inspector I'd say that this is even worse. The inspector does not check for the interface at all, but only greps the value from the path. If this method is called with the wrong document, it returns something wrong instead of failing.